django_project/lesson/views/worksheet.py

```diff
@@ -218,6 +218,56 @@ def render_to_response(self, context, **response_kwargs):
         return zip_response
 
 
+class WorksheetDownloadSampleDataView(WorksheetDetailView):
+    def render_to_response(self, context, **response_kwargs):
+        context = self.get_context_data()
+        file_title = (f"{context['worksheet'].section.name}"
+                      f"-{context['worksheet'].module}")
+        s = BytesIO()
+        zf = zipfile.ZipFile(s, "w")
+
+        if context['worksheet'].external_data:
+            data_path = context['worksheet'].external_data.url
+            zip_data_path = settings.MEDIA_ROOT + data_path[6:]
+
+            try:
+                external_file_zf = zipfile.ZipFile(zip_data_path)
+                for name in external_file_zf.namelist():
+                    if name.endswith('.zip'):
+                        try:
+                            filebytes = BytesIO(external_file_zf.read(name))
+                            subzip = zipfile.ZipFile(filebytes)
+                            for name_subzip in subzip.namelist():
+                                if name_subzip.endswith('/'):
+                                    continue
+                                if name_subzip.startswith('__MACOSX'):
```

Comment:
@sumandari what about other hidden files? LGTM for now, I think in the future it's better to create a ignore list file.

Reply:
thank you @dimasciput 
I will update this after https://github.com/kartoza/prj.app/pull/1314 reviewed/merged. I will use the same ignore list from it.

Reply:
LGTM thanks

Reply:
I will merge this for now, please see #1382 for ignore list file